java-checks/src/main/java/org/sonar/java/checks/ConstantsShouldBeStaticFinalCheck.java

```diff
@@ -87,23 +86,28 @@ public void leaveNode(Tree tree) {
   private static boolean hasConstantInitializer(VariableTree variableTree) {
     ExpressionTree init = variableTree.initializer();
     if (init != null) {
```

Comment:
I think an early return (`if (init == null) {return;}`) might be better here, as it would reduce the nesting level of the inner block (cognitive complexity), and also, the contents of a nested block should not be too long.